### PR TITLE
Document EnTanron5 (destructible props, fragments, and item drops in Twinmold's arena)

### PR DESCRIPTION
**Pull request by tom-overton**
Documenting this actor was trickier than I expected it to be. There is probably more documentation or cleanup I could do, but at this point, I figured it would be more productive to toss this out into the world and get feedback than try to do it all myself.

I also changed some things in Boss02 while I was working on this. Specifically, I added a `TWINMOLD_GET_TYPE` macro to use of using a raw `params` access, I named `sGiantModeScaleFactor`, I added `TYPE` to the types enum, and I cleaned up some stuff involving effects.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:379]:
```suggestion
                        fragmentAndDustPos.x = pos.x + (Rand_CenteredFloat(xzFactor) * sGiantModeScaleFactor);
                        fragmentAndDustPos.z = pos.z + (Rand_CenteredFloat(xzFactor) * sGiantModeScaleFactor);
                        fragmentAndDustPos.y = this->actor.floorHeight + (yFactor * sGiantModeScaleFactor);
```
I think this reads a little easier? (it matches)

**Review by engineer124 — COMMENTED** [review]
Really nice work. Just some small comments

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:13]: I was sort of thinking of doing something like this, but I didn't know if `PROPS` was appropriate. Would you call the item drops or the fragments "props"?

Separately from that, I was thinking of calling this `PILLAR_RUIN` instead. That way, it's more clear that it's roughly the same kind of thing as the pyramid ruin. I could also do `RUIN_PILLAR` and `RUIN_PYRAMID` to put the fact that they're ruins up-front. What do you think?

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:52]: I was wondering about the wording, but looking in the current codebase, `rotationalVelocity` is used in multiple places, and `angularVelocity` is never used. And looking at code like this:
```c
        this->actor.shape.rot.x += this->ruinFragmentRotationalVelocityX;
        this->actor.shape.rot.y += this->ruinFragmentRotationalVelocityY;
```
Since it's changing a rotation value, it sort of makes sense to call it `rotationalVelocity`.

Comment (hensldm) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:53]: `angularVelocity` is the same as `rotationalVelocity`, though `angularVelocity` is usually the term used in Physics. Personally since we are already using `rotationalVeclocity` lets use that and possibly change it in the future (I would also argue shortening to vel as well when the time comes)

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h
Comment (hensldm) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:14]: Personally the fragments seem more like `props` to me. Though I like the idea of `RUIN_*` for both of them.

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (hensldm) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:44]:
```suggestion
s32 sRuinFragmentItemDropCount = 0;
```
Along with the variable in `variables.txt`

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (hensldm) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:85]:
```suggestion
 * Stores the X and Z spawn positions for all pillars and ruins. Their Y spawn position is determined
```

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h
Comment (hensldm) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:14]: Also want to add that it seems in most comments you already use prop to refer to just the `PILLAR`s and `PYRAMID`s

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:53]: Yeah, that sounds like a plan to me. I can even make the switch to `angularVelocity` as soon as this PR is merged; I'll just do it all at once.

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:14]: Yeah, you two have convinced me that these are all props, so I'll make the switch over. I'll have to rephrase some comments, but that's fine.

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:93]:
```suggestion
    { 1200, 2500 },  // TWINMOLD_PROP_TYPE_RUIN_PILLAR_1 
    { -1200, 2500 },  // TWINMOLD_PROP_TYPE_RUIN_PILLAR_2
    { 1200, -2500 },  { -1200, -2500 }, { 2500, 1200 },
    { -2500, 1200 }, { 2500, -1200 },  { -2500, -1200 }, { 1000, 1000 },   { -1000, 1000 },
    { 1000, -1000 }, { -1000, -1000 }, { 0, -1000 },     { 0, 1000 },      { 1000, 0 },
    { -1000, 0 },    { 0, -2000 },     { 0, 2000 },      { 2000, 0 },      { -2000, 0 },
```
Could you add the types to these? and also `sDLists` and `sBaseScales`?

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:100]: Should `MajoraPillar` be updated to the new type name `RuinPillar`?

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:537]:
```suggestion
                        Item_Give(play, ITEM_MAGIC_JAR_BIG);
```
I'm surprised no conflicts were detected, but this will be updated once master is merged. Could you update:
`TWINMOLD_PROP_ITEM_DROP_TYPE_LARGE_MAGIC` -> `TWINMOLD_PROP_ITEM_DROP_TYPE_MAGIC_JAR_BIG`
as well as the file description:
`player a large amount of magic`.

Looking at this file also makes me realize `gDropMagicLargeTex` needs updating, but I can do that separately in the next cleanup

Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:537]: I merged `master` and it caused no issues. I think it's because git knows I haven't changed this line, so it automatically do the merge for me.

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:93]: Are you sure this is really a good idea? This would increase the number of lines that this data takes up from 11 to 60. Maybe for this position one, it might be worth it, but for the other two, I'm not sure it adds much. Every single value in `sBaseScales` is identical, and the type used for each entry in `sDLists` is fairly obvious.

Comment (engineer124) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:93]: I personally still like it with the comments, no matter the num of lines it takes, since eventually these params will go into the scene spawn info, and it would make it much faster to search which pillar is mapped to which position/scale/dList, and if you were to add a pillar, it'd be easier to tell where to shift depending on where you add an enum for example. It just makes it easier to work with.

You could also just add a comment splitting it in 2 parts (the ruin pillars and the ruin pyramids)

Thoughts?

Comment (tom-overton) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:93]: Yeah, the modability argument is fair. It just felt like such a waste to expand `sBaseScales` any more than it needs to be, since it's basically useless. But I suppose if you wanted to actually *use* this, you would like to know which number corresponds to what ruin.

By the way, I *don't* think these params will ever move to the scene spawn info, because these are never actually spawned as part of the scene. Boss02 spawns the static instance of EnTanron5, which then spawns all the rest of them.

**Review by hensldm — APPROVED** [review]

File: assets/xml/objects/object_boss02.xml
Comment (engineer124) [inline assets/xml/objects/object_boss02.xml:45]: Update OutNames? PR LGTM after that

Comment (tom-overton) [reply assets/xml/objects/object_boss02.xml]: Oops, good catch. Fixed.

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**
